core/server/worker/src/main/java/alluxio/worker/block/UfsIOManager.java

```diff
@@ -0,0 +1,298 @@
+   * @param user the client name or tag
+   * @param throughput throughput limit, MB/s
+   */
+  public void setQuota(String user, int throughput) {
+    Preconditions.checkArgument(throughput > 0, "throughput should be positive");
+    mThroughputQuota.put(user, throughput);
+  }
+
+  private void schedule() {
+    while (!Thread.currentThread().isInterrupted()) {
+      ReadTask task = null;
+
+      try {
+        task = mReadQueue.take();
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+      }
+      if (task != null) {
+        int quota = mThroughputQuota.getOrDefault(task.getUser(), -1);
+        double rate = getUsedThroughput(task.mUser);
+        if (quota == -1 || quota * Constants.MB > rate) {
+          try {
+            mUfsIoExecutor.submit(task);
+          } catch (RejectedExecutionException e) {
+            // should not reach here since we use unbounded queue in thread pool and use a bounded
+            // blocking queue upfront to control number of read tasks
+            mReadQueue.add(task);
+          }
+        } else { // resubmit to queue
+          mReadQueue.add(task);
+        }
+      }
+    }
+  }
+
+  @VisibleForTesting
+  public double getUsedThroughput(String user) {
+    Meter ufsBytesReadThroughput = mUfsBytesReadThroughputMetrics.computeIfAbsent(
+        new BytesReadMetricKey(mUfsClient.getUfsMountPointUri(), user),
+        key -> key.getUser() == null
+            ? MetricsSystem.meterWithTags(MetricKey.WORKER_BYTES_READ_UFS.getName(),
+            MetricKey.WORKER_BYTES_READ_UFS.isClusterAggregated(), MetricInfo.TAG_UFS,
+            MetricsSystem.escape(key.getUri()))
+            : MetricsSystem.meterWithTags(MetricKey.WORKER_BYTES_READ_UFS.getName(),
+            MetricKey.WORKER_BYTES_READ_UFS.isClusterAggregated(), MetricInfo.TAG_UFS,
+            MetricsSystem.escape(key.getUri()), MetricInfo.TAG_USER, key.getUser()));
+    return ufsBytesReadThroughput.getOneMinuteRate();
+  }
+
+  /**
+   * read from ufs.
+   * 
+   * @param blockId block id
+   * @param blockSize block size
+   * @param ufsPath ufs path
+   * @param offset offset to read
+   * @param length length to read
+   * @param positionShort is position short or not
+   * @param user user tag
+   * @return content read
+   * @throws ResourceExhaustedException any exception from ufs
+   */
+  public CompletableFuture<byte[]> read(long blockId, long blockSize, String ufsPath, long offset,
+      long length, boolean positionShort, String user) throws ResourceExhaustedException {
+    CompletableFuture<byte[]> future = new CompletableFuture<>();
+    if (mReadQueue.size() >= READ_CAPACITY) {
+      throw new ResourceExhaustedException(format("UFS read at capacity, try later or increase %s",
+          PropertyKey.UNDERFS_IO_READ_QUEUE_CAPACITY.getName()));
+    }
+    mReadQueue.add(
+        new ReadTask(blockId, blockSize, ufsPath, offset, length, positionShort, user, future));
```

Comment:
The logic would be simpler / clearer if you put the metric into the `ReadTask` as well. If nothing else, you don't need to have the long `computeIfAbsent` code block at multiple places.